### PR TITLE
Time to remove LISP, overlay, and underlay docs and names

### PR DESCRIPTION
**Pull request by eriknordmark**
A long time ago we had integrated LISP secure overlay network code, and as a result zedrouter had functions and variables named "underlay*" or "ul*".

I noticed we still have some text left in SECURITY.md so I did a pass over the code.

Note that there are several commits with different potential for merge conflicts with existing work; the "underlay" commit with "ul*"  renames is the one most likely to cause such issues. Note that the Underlay struct is renamed to AppNetAdapter{Config,Status} but I'm open to other name suggestions.

The underlay term still remains for the geolocation code and API, but added comments explaining that this is the underlay/public IP address.

Let me know if this is a good time to do this cleanup or if it will cause conflicts with ongoing network work and should be deferred.

### CODE REVIEW

File: pkg/pillar/types/zedroutertypes.go [hunk 1]
```diff
@@ -137,13 +137,13 @@ func (status AppNetworkStatus) AwaitingNetwork() bool {
...
-// GetULStatusForNI returns UnderlayNetworkStatus for every application VIF
+// GetULStatusForNI returns AppNetAdapterStatus for every application VIF
 // connected to the given network instance (there can be multiple interfaces connected
 // to the same network instance).
-func (status AppNetworkStatus) GetULStatusForNI(netUUID uuid.UUID) []*UnderlayNetworkStatus {
-	var uls []*UnderlayNetworkStatus
-	for i := range status.UnderlayNetworkList {
-		ul := &status.UnderlayNetworkList[i]
+func (status AppNetworkStatus) GetULStatusForNI(netUUID uuid.UUID) []*AppNetAdapterStatus {
+	var uls []*AppNetAdapterStatus
```
Comment (milan-zededa) [inline pkg/pillar/types/zedroutertypes.go:144]: I think we should also rename `uls` and `ul` variables (there are several occurrences of them across the code base)

Comment (milan-zededa) [reply pkg/pillar/types/zedroutertypes.go:144]: Perhaps `uls`->`adapters`, `ul` -> `adapter`?

File: pkg/pillar/types/zedroutertypes.go [hunk 2]
```diff
@@ -47,14 +47,14 @@ func (e *DNSNotAvail) Error() string {
...
-	GetStatsIPAddr      net.IP
-	UnderlayNetworkList []UnderlayNetworkConfig
-	CloudInitUserData   *string `json:"pubsub-large-CloudInitUserData"`
-	CipherBlockStatus   CipherBlockStatus
-	MetaDataType        MetaDataType
+	UUIDandVersion    UUIDandVersion
+	DisplayName       string
+	Activate          bool
+	GetStatsIPAddr    net.IP
+	AppNetAdapterList []AppNetAdapterConfig
```
Comment (milan-zededa) [inline pkg/pillar/types/zedroutertypes.go:54]: What about using the plural `AppNetAdapters` and avoid `List` at the end?

**Comment by milan-zededa** [comment]
I think we can go ahead with this - I will rebase https://github.com/lf-edge/eve/pull/3513 and fix merge conflicts once this is merged.

**Comment by milan-zededa** [comment]
Note that some workflows are currently failing because of this issue: https://github.com/golang/go/issues/63684

**Review by uncleDecart — APPROVED** [review]
LGTM, +1 to change names

File: pkg/pillar/types/zedroutertypes.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/types/zedroutertypes.go:144]: I used ana elsewhere but missed this. adapters makes sense here.
Do you think adapter is more clear elsewhere where I'm using ana?

Comment (milan-zededa) [reply pkg/pillar/types/zedroutertypes.go:144]: I think that I would prefer adapter, but ana is fine too.

Comment (milan-zededa) [reply pkg/pillar/types/zedroutertypes.go]: Yes, this was just copied and can be removed.

**Review by shjala — APPROVED** [review]

**Review by milan-zededa — APPROVED** [review]

**Pull request merged**